---
title: "Class17"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Cancer Genomics

## Installed and load the packages
```{r}
#source("https://bioconductor.org/biocLite.R")
#biocLite(c("GenomicDataCommons", "maftools"))
library(GenomicDataCommons)
library(maftools)
```

##Check on GDC status
```{r}
GenomicDataCommons::status()
```

```{r}
cases_by_project <- cases() %>%
  facet("project.project_id") %>%
  aggregations()
head(cases_by_project)
```

## Barplot of the cases per project. Log scale for the y axis (log="y"), rotated axis labels (las=2) and color the bar coresponding to the TCGA-PAAD project.
```{r}
x <- cases_by_project$project.project_id

# Make a custom color vector for our plot
colvec <- rep("lightblue", nrow(x))

colvec[x$key == 'TCGA-PAAD'] <- "red"

# Plot with 'log' for y axis and rotate labels with 'las'
par(mar = c(10, 4, 4, 2) + 0.1)  
barplot(x$doc_count, names.arg=x$key, log='y', col=colvec, las=2)
```

## Find all gene expression data files for all pancreatic cancer patients
```{r}
# This code snipet is taken from the package vignette
file_records <- files() %>%
  filter(~ cases.project.project_id == "TCGA-PAAD" &
    data_type == "Gene Expression Quantification" &
    analysis.workflow_type == "HTSeq - Counts") %>%
  response_all()
```

```{r}
#View(file_records)
# Number of rows of RNA-Seq data from the ‘TCGA-PAAD’ project
nrow(file_records$results)
```

##Find MAF format somatic mutation files for our ‘TCGA-PAAD’ project. Focus on only one of the MAF files for this project in GDC, namely the MuTect2 workflow variant calls
```{r}
maf.files = files() %>%
    filter(~ cases.project.project_id == 'TCGA-PAAD' &
        data_type == 'Masked Somatic Mutation' &
        data_format == "MAF" &
        analysis.workflow_type == "MuTect2 Variant Aggregation and Masking"
    ) %>%
    response_all()
```

```{r}
#View(maf.files)
attributes(maf.files)
```

## Use the ids() function to pull out the unique identifier for our MAF file
```{r}
uid <- ids(maf.files)
uid
#maffile = gdcdata(uid, destination_dir =".")
maffile = gdcdata(uid)

maffile
```

```{r}
vars = read.maf(maf = maffile, verbose = FALSE)

plotmafSummary(maf =vars, rmOutlier = TRUE,
               addStat = 'median', dashboard = TRUE,
               titvRaw = FALSE)

```

```{r}
oncoplot(maf = vars, top = 10)
```

```{r}
# Oncoplot for our top 10 most frequently mutated genes
pdf("oncoplot_panc.pdf")
oncoplot(maf = vars, top = 10, fontSize = 12)
dev.off()
```

```{r}
oncostrip(maf=vars, genes=c("KRAS", "TP53"))
```

```{r}
lollipopPlot(maf = vars, gene = 'KRAS', 
                         showMutationRate = TRUE, domainLabelSize = 3)

```

```{r}
lollipopPlot(maf = vars, gene = 'TP53')
```
